packages/fx-core/src/common/deps-checker/internal/funcToolChecker.ts

```diff
@@ -1,517 +1,348 @@
-import {
-  DepsCheckerError,
-  LinuxNotSupportedError,
-  GlobalFuncNodeNotMatchedError,
-  NodeNotFoundError,
-  PortableFuncNodeNotMatchedError,
-} from "../depsError";
-import { cpUtils } from "../util/cpUtils";
-import { isLinux, isWindows } from "../util/system";
-import { DepsCheckerEvent, TelemtryMessages } from "../constant/telemetry";
+import { getLocalizedString } from "../../localizeUtils";
+import { defaultHelpLink, v3NodeNotFoundHelpLink } from "../constant/helpLink";
+import { Messages } from "../constant/message";
+import { DependencyStatus, DepsChecker, DepsType, FuncInstallOptions } from "../depsChecker";
+import { DepsCheckerError, LinuxNotSupportedError, NodeNotFoundError } from "../depsError";
 import { DepsLogger } from "../depsLogger";
 import { DepsTelemetry } from "../depsTelemetry";
-import { DepsChecker, DependencyStatus, DepsType, FuncInstallOptions } from "../depsChecker";
-import { Messages } from "../constant/message";
+import { cpUtils } from "../util/cpUtils";
+import { createSymlink, rename, unlinkSymlink } from "../util/fileHelper";
+import { isLinux, isWindows } from "../util/system";
 import { NodeChecker } from "./nodeChecker";
-import { getLocalizedString } from "../../localizeUtils";
-
-export enum FuncMajorVersion {
-  v1 = "1",
-  v2 = "2",
-  v3 = "3",
-  v4 = "4",
-}
-
-const FuncNodeVersionWhiteList: { [key: string]: { [key: string]: boolean } } = {
-  // func-core-tools version
-  "3": {
-    "10": true, // node version
-    "12": true,
-    "14": true,
-  },
-  "4": {
-    "14": true,
-    "16": true,
-    "18": true,
-  },
-};
-
-const RecommendedNodeVersion = "16";
 
-interface FuncVersion {
-  majorVersion: FuncMajorVersion;
+type FuncVersion = {
+  majorVersion: number;
   minorVersion: number;
   patchVersion: number;
-}
-
-const MinNode18FuncVersion: FuncVersion = {
-  majorVersion: FuncMajorVersion.v4,
-  minorVersion: 0,
-  patchVersion: 4670,
-};
-
-type FuncInstallationStatus = {
-  isInstalled: boolean;
-  funcVersion: FuncMajorVersion | null;
+  versionStr: string;
 };
 
 const funcPackageName = "azure-functions-core-tools";
 const funcToolName = "Azure Functions Core Tools";
 
-const installVersion = FuncMajorVersion.v4;
-const supportedVersions = [FuncMajorVersion.v4];
-const displayFuncName = `${funcToolName} (v${FuncMajorVersion.v4})`;
-
 const timeout = 5 * 60 * 1000;
 
 export class FuncToolChecker implements DepsChecker {
-  private readonly _logger: DepsLogger;
-  private readonly _telemetry: DepsTelemetry;
-
-  constructor(logger: DepsLogger, telemetry: DepsTelemetry) {
-    this._logger = logger;
-    this._telemetry = telemetry;
+  private log: string;
+  constructor(logger?: DepsLogger, telemetry?: DepsTelemetry) {
+    // TODO: add log and telemetry
+    this.log = "";
   }
 
   public async getDepsInfo(
-    isPortableFuncInstalled: boolean,
-    isGlobalFuncInstalled: boolean,
+    funcVersion: FuncVersion | null,
```

Comment:
`| undefined`?

Reply:
Or, why `null` instead of `undefined`?

Reply:
The old code use `private async queryFuncVersion(path: string): Promise<FuncVersion | null>` , which returned a `FuncVersion | null` object. This return value was then passed to all the upper method. 
We can refactor the code, may be in another PR.